javav2/example_code/ec2/src/test/java/EC2Test.java

```diff
@@ -33,172 +38,190 @@ public class EC2Test {
+        System.out.println("\n Test 4 passed");
     }
 
     @Test
+    @Tag("IntegrationTest")
     @Order(5)
-    public void DeleteKeyPair() {
-         DeleteKeyPair.deleteKeys(ec2,keyName);
-         System.out.println("\n Test 5 passed");
-    }
+    public void CreateSecurityGroup() {
+        groupId = CreateSecurityGroup.createEC2SecurityGroup(ec2,groupName,groupDesc,vpcId);
+        assertFalse(groupId.isEmpty());
+        System.out.println("\n Test 5 passed");
+   }
 
     @Test
+    @Tag("IntegrationTest")
     @Order(6)
-    public void CreateSecurityGroup() {
-       groupId = CreateSecurityGroup.createEC2SecurityGroup(ec2,groupName,groupDesc,vpcId);
-       System.out.println("\n Test 6 passed");
-   }
+    public void DescribeSecurityGroup() {
+        assertDoesNotThrow(() -> DescribeSecurityGroups.describeEC2SecurityGroups(ec2,groupId));
+        System.out.println("\n Test 6 passed");
+    }
 
     @Test
+    @Tag("IntegrationTest")
     @Order(7)
-    public void DescribeSecurityGroup() {
-        DescribeSecurityGroups.describeEC2SecurityGroups(ec2,groupId);
+    public void DeleteSecurityGroup(){
+        assertDoesNotThrow(() -> DeleteSecurityGroup.deleteEC2SecGroup(ec2, groupId));
         System.out.println("\n Test 7 passed");
     }
 
 
     @Test
+    @Tag("IntegrationTest")
     @Order(8)
-    public void DeleteSecurityGroup(){
-        DeleteSecurityGroup.deleteEC2SecGroup(ec2, groupId);
+    public void DescribeAccount() {
+        assertDoesNotThrow(() -> DescribeAccount.describeEC2Account(ec2));
         System.out.println("\n Test 8 passed");
     }
 
-
     @Test
+    @Tag("IntegrationTest")
     @Order(9)
-    public void DescribeAccount() {
-        DescribeAccount.describeEC2Account(ec2);
+    public void DescribeInstances() {
+        assertDoesNotThrow(() -> DescribeInstances.describeEC2Instances(ec2));
         System.out.println("\n Test 9 passed");
     }
 
     @Test
+    @Tag("IntegrationTest")
     @Order(10)
-    public void DescribeInstances() {
-        DescribeInstances.describeEC2Instances(ec2);
+    public void DescribeRegionsAndZones () {
+        assertDoesNotThrow(() -> DescribeRegionsAndZones.describeEC2RegionsAndZones(ec2));
         System.out.println("\n Test 10 passed");
     }
 
     @Test
+    @Tag("IntegrationTest")
     @Order(11)
-    public void DescribeRegionsAndZones () {
-      DescribeRegionsAndZones.describeEC2RegionsAndZones(ec2);
-      System.out.println("\n Test 11 passed");
+    public void DescribeVPCs () {
+        assertDoesNotThrow(() -> DescribeVPCs.describeEC2Vpcs(ec2,vpcId));
+        System.out.println("\n Test 11 passed");
     }
 
     @Test
+    @Tag("IntegrationTest")
     @Order(12)
-    public void DescribeVPCs () {
-      DescribeVPCs.describeEC2Vpcs(ec2,vpcId);
-      System.out.println("\n Test 12 passed");
-    }
+   public void FindRunningInstances() {
+        assertDoesNotThrow(() -> FindRunningInstances.findRunningEC2Instances(ec2));

```

Comment:
A lot of these tests could be more robust that testing that the functions don't throw an error. For instance, does this actually return instances? Are those instances running? What if the results are paginated? This test doesn't answer those questions.

Reply:
That was the original objective of these tests - make they they work and do not throw exceptions.

Reply:
I'm not rejecting the PR over it, just an observation.